### ISSUE TITLE
Wrong return type on a generic function that takes an array of maps

### ISSUE THREAD

**Issue by pdesaulniers** [teal-language/tl#111]
```lua
local MyRecord = record
    func: function<K, V>(t: {{K:V}}): {V}
end

local x: {string} = MyRecord.func({
    {id = 2}, 
    {otherkey = "hello"}
})
```
```
tl check test.tl
```

```
========================================
1 error:
test.tl:5:7: in local declaration: x: got {number}, expected {string}
```

In this example, `x` should have a type of `{number | string}`.

**Comment by pdesaulniers** [comment]
If I had another item to the sequence, the compiler crashes:
```lua
local x: {string} = MyRecord.func({
    {id = 2}, 
    {id = "hello"},
    {id = "yo"}
})
```

```
lua: .//tl.lua:4880: attempt to index a nil value (field 'ktype')
stack traceback:
        .//tl.lua:4880: in upvalue 'expand_type'
        .//tl.lua:5500: in field 'after'
        .//tl.lua:2064: in function <.//tl.lua:2058>
        (...tail calls...)
        .//tl.lua:2152: in upvalue 'recurse_node'
        .//tl.lua:2249: in upvalue 'recurse_node'
        .//tl.lua:2152: in upvalue 'recurse_node'
        .//tl.lua:2159: in upvalue 'recurse_node'
        .//tl.lua:2152: in upvalue 'recurse_node'
        .//tl.lua:6014: in function 'tl.type_check'
        .//tl.lua:6123: in function 'tl.process'
        ./tl:210: in main chunk
        [C]: in ?
```

**Comment by hishamhm** [comment]
Fixed! If you look at 8167f3e, this is a bug that tagged unions (or equivalent) would have caught...
